### DISCUSSION TITLE
Understanding the CSV export from the "Coverage" page in GHAS

### DISCUSSION THREAD

**Discussion by xepiercex** [community/community#75681] [Code Security] [answered]
### Select Topic Area

Question

### Body

I have been trying to build out team-specific metrics for SAST coverage based on the CSV export you can download with this process here:  https://docs.github.com/en/enterprise-cloud@latest/code-security/security-overview/exporting-data-from-the-risk-and-coverage-pages

When I do this, there are a number of repos flagged as "Ineligible" under a column called "Code Scanning Default Setup Status".  I have not been able to find any documentation that clearly explains what this means.  My _guess_ is that either the majority language or none of the languages in the codebase are natively supported by CodeQL.  Does anyone know for sure or can point me to a document that actually confirms this and possibly other default fields on the CSV export?  Are there other reasons a repo could be flagged as ineligible?  Does this leave open the possibility that "ineligible" codebases can still be scanned by a third party scanner?  Thanks.

**Comment by chrisgavin** [comment, marked as answer]
Hi @xepiercex. The reasons why a repository might be considered ineligible for CodeQL default setup are [listed here](https://docs.github.com/en/enterprise-cloud@latest/code-security/code-scanning/enabling-code-scanning/configuring-default-setup-for-code-scanning#requirements-for-using-default-setup). Indeed the most common is that the repository does not have any languages supported by CodeQL. In this case you can use a third-party tool with support for scanning the languages you are interested in.

**Reply by xepiercex** [reply]
Thank you, Chris, for the confirmation and the link!
